### ISSUE TITLE
library is incompatible with Flash Builder 4.5

### ISSUE THREAD

**Issue by xchg-dot-ca** [mikechambers/as3corelib#171]
After using as3corelib in my project getting the following message when trying to edit UI in Design Mode:

D:..Projects\Client\libs\as3corelib.swc- DesignAssetLoader.CompleteTimeoutSWC file failed to load. Any component dependent on this SWC file will not be displayed in the Design Mode.The SWC may have failed to load because of:   
-  Incompatible definitions caused by usage of a different SDK version  
-  Missing referred class definitions 

Flash builder 4.5.1, Flex SDK 4.5.1, as3corelib .93

Very easy to reproduce, create new Flex Desktop project, call it Test, add as3corelib in libs, open Test.mxml in Design mode... done, you have an error

**Comment by mikechambers** [comment]
Yeah. The SWC hasnt been tested with 4.5. I would suggest that you try
to link the source, instead of the SWC.

Hope that helps...

mike

On Tue, Jul 19, 2011 at 8:43 PM, riahut-dot-com
reply@reply.github.com
wrote:

**Comment by npvathi** [comment]
I created a new flex library for as3corelib and added the src to my flex desktop project (FlashBuilder sdk: 4.5.1, win) and it says it has a problem for recognizing JSON. I am trying out the example Json as specified in the examples folder. 

Description Resource    Path    Location    Type
1120: Access of undefined property JSON.  
Can not resolve a multiname reference unambiguously. JSON (from C:...\Adobe Flash Builder 4.5\sdks\4.5.1\frameworks\libs\air\airglobal.swc(JSON, Walker)) and com.adobe.serialization.json:JSON (from C:\Users\xxx\Downloads\Json\as3corelib-.93\lib\as3corelib.swc(com.adobe.serialization.json:JSON)) are available.

**Comment by DjTriple7** [comment]
Did you ever find a resolution to this? I am having the same issue with Flash Builder 4.5.1 w/AIR 3 on OS X (using the source, not SWC).

Thanks!

**Comment by jwb** [comment]
If you're using AIR, you probably want the JSON class that's included in 4.5.1. Use the standard JSON.stringify and JSON.parse instead of the as3corelib implementation which doesn't implement the ECMA standard. I'm gonna post a pull request with a version that deprecates the ad hoc API and implements a compatible subset of the ECMA spec.

**Comment by muni2explore** [comment]
i had found the following when i importing as3 api files into flex 4.6 development environment the following error

Access of undefined property JSON.
Access of undefined property JSON.
Cannot resolve a multiname reference unambiguously json

find the solution for the above error in the following URL

http://www.smarttutorials.net/cannot-resolve-a-multiname-reference-unambiguously-json-in-flash-builder-4-6-with-smartfox-server/

**Comment by ghost** [comment]
I think most of you might have already figured out the solution for this by now. I am writing this to help out new comers to FB.

The "Cannot resolve a multiname reference unambiguously json" problem occurs since the new sdk has an AIR service defined with the same name. The easiest work around for this is to change the JSON from all uppercase to all lowercase letters and you are good to go. 

ex:
change this - com.adobe.serialization.json.JSON
to this - com.adobe.serialization.json.json (Rename the 'JSON.as' to 'json.as' and change it in the class name)

Hope this helps to those who are looking fro this. :+1:
